libs/shared/rank/src/lib/shared-rank.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
+import { Box } from '@material-ui/core';
+import { Rating } from '@material-ui/lab';
+
+import './shared-rank.scss';
+
+export interface SharedRankProps {
+  value: number;
+  isRanked: boolean;
+  onUpdate: (data) => void;
+}
+
+export const SharedRank = (props: SharedRankProps) => {
+  const [rank, setRank] = useState({
+    isRanked: false,
+    value: null,
+  });
+
+  useEffect(() => {
+    setRank({ ...props });
+  }, [props]);
+
+  return (
+    <div className="rank">
+      <Rating
+        disabled={rank.isRanked}
+        name="hover-feedback"
+        value={rank.value}
+        precision={0.5}
+        onChange={(event: EventTarget, newValue: number) => {
```

Comment:
Please fix this warning 
![image](https://user-images.githubusercontent.com/36937039/87696528-d5858b00-c788-11ea-8847-38115336903d.png)